Ft added database

anothervoyager opened the pull request.

anothervoyager committed:
feat: добавлено взаимодействие с базой данных
feat: Небольшие изменения в выводе ошибок - принты были заменены на raise ValueError

anothervoyager committed:
fix: удален task_manager.py за ненадобностью, весь функционал перенесен в database.py

anothervoyager committed:
Feat: созданы миграции alembic

anothervoyager committed:
docs: дополненна документация кода досктрингами

Rognak commented on .idea/FT-added-databse.iml:
Залетели .idea файлы, необходимо убрать через .gitignore

Rognak commented on alembic/versions/b3d30a2e9d64_initial_migration.py:
```diff
@@ -0,0 +1,32 @@
+def upgrade() -> None:
+    """Upgrade schema."""
+    # ### commands auto generated by Alembic - please adjust! ###
+    pass
```
Пустая миграция.

Rognak commented on database.py:
```diff
@@ -0,0 +1,121 @@
+Session = sessionmaker(bind=engine)
+
+
+class TaskModel(Base):
```
Модели БД должны быть в отдельном файле.

Rognak commented on database.py:
```diff
@@ -0,0 +1,121 @@
+
+        Создает новую сессию для выполнения операций с базой данных.
+        """
+        self.session = Session()
```
Сильная связь. Передавай сессию как аргумент инициализатора

Rognak commented on database.py:
```diff
@@ -0,0 +1,121 @@
+        """
+        task_model = self.session.query(TaskModel).filter(TaskModel.task_id == task_id).first()
+        if task_model:
+            task_model.completed = 'Y'
```
Давай использовать bool для полей, где подразумевается "да/нет"

Rognak commented on main.py:
```diff
@@ -0,0 +1,93 @@
+        """
+        Инициализирует объект базы данных, создавая экземпляр DatabaseManager.
+        """
+        self.db_manager = DatabaseManager()
```
Аналогично

Rognak commented on task.py:
```diff
@@ -0,0 +1,46 @@
+class Task:
```
Тогда должен быть уже ConsoleManager

Rognak commented on main.py:
```diff
@@ -0,0 +1,93 @@
+    Основной класс приложения Todo, отвечающий за управление задачами.
+    """
+
+    def __init__(self):
```
```python
def __init__(self, manager: IManager):
    self.manager = manager
```

Rognak replied:
+ Необходимо добавить интерфейс самого менеджера
```python
class IManager(abc):
   ...
```

Rognak reviewed (changes requested):
Необходимо исправить замечания

anothervoyager committed:
fix: пофикшены пустые миграции, новая миграция recreate_tasks_table.
fix: убрал жесткую связь, логика выполнения команд вынесена в отдельные классы
fix: Вместо создания экземпляра DatabaseManager внутри TodoApp, он теперь создается вне класса и передается как параметр в конструктор. аналогично с session.
fix: модели выделены в отдельный файл.

anothervoyager committed:
fix: пофикшены пустые миграции, новая миграция recreate_tasks_table.
fix: убрал жесткую связь, логика выполнения команд вынесена в отдельные классы
fix: Вместо создания экземпляра DatabaseManager внутри TodoApp, он теперь создается вне класса и передается как параметр в конструктор. аналогично с session.
fix: модели выделены в отдельный файл.
fix: tasks перименован в consolemanager

anothervoyager committed:
Удалена папка .idea из репозитория

anothervoyager committed:
Удалена папка .idea из репозитория и добавлена в .gitignore

anothervoyager committed:
Delete .idea directory

anothervoyager committed:
fix: исправлено отображение выполненных задач при вводе команды get-tasks

anothervoyager committed:
fix: исправлены замечания, добавлен интерфейс IManager и два класса который реализуют IManager - ConsoleManager и DataBaseManager, переделан класс TodoApp
